hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -347,11 +347,36 @@ func (r *NodePoolReconciler) updatingConfigCondition(ctx context.Context, nodePo
 	targetConfigHash := token.HashWithoutVersion()
 	isUpdatingConfig := isUpdatingConfig(nodePool, targetConfigHash)
 	if isUpdatingConfig {
+		reason := hyperv1.AsExpectedReason
+		message := fmt.Sprintf("Updating config in progress. Target config: %s", targetConfigHash)
+		status := corev1.ConditionTrue
+
+		if nodePool.Spec.Management.UpgradeType == hyperv1.UpgradeTypeInPlace {
+			machineSet, err := r.getMachineSet(ctx, nodePool)
```

Comment:
instead of creating this new helper, please build a capi resource via
```newCAPI(token, infraID)```
then you can get the machineset object key capi.machineSet(), so then you can get it here